android/java/org/chromium/chrome/browser/crypto_wallet/util/TokenUtils.java

```diff
@@ -66,10 +67,13 @@ private static BlockchainToken[] filterTokens(NetworkInfo selectedNetwork,
                 default:
                     throw new UnsupportedOperationException("Token type not supported.");
             }
-            return typeFilter || isSameToken(t, nativeAsset) || (keepVisibleOnly && !t.visible);
+            return typeFilter || (includeNativeAsset && isSameToken(t, nativeAsset))
```

Comment:
If `includeNativeAsset` is `false`, then native assets will not be removed from the list, which is not desired I think?

Reply:
In that case what does `includeNativeAsset` mean? It seems to me that what happens is (1) if `includeNativeAsset` is true, then the native asset is removed and re-added to front; (2) if `includeNativeAsset` is false then native asset is not removed.

Reply:
it means does the native asset has to be included in the given list or not.

Reply:
we had a situation that we had an empty list and we added a native asset in it, but we shouldn't do that as it's not supposed to allow a purchase of it.

Reply:
I actually think that we don't need to call `filterTokens` at all for buy case. As it does basically nothing, we receive a complete list of tokens from a backend that are allowed to be purchased or an empty list if there are none, why would we filter them? wdyt @simoarpe @Pavneet-Sing ?

Reply:
But isn't that when `includeNativeAsset` is false then the native asset is not removed (but we wish to remove it)?

Reply:
Ok I see. It is getting too confusing for the function. If we can avoid adding the parameter then we should do it.

Reply:
I reverted the function to be without the extra parameter and removed the call to `filterTokens` as it's useless for a buy case and adds a native token that we don't want.

Reply:
#Re: https://github.com/brave/brave-core/pull/17713#discussion_r1145320370
Make sense to remove this for checking when buy should be enabled or not, 
though the another use-case is showing buy token list in `EditVisibleAssetsBottomSheetDialogFragment`: https://github.com/brave/brave-core/blob/d4a38815811838182d2609a793499e68d3055db2/android/java/org/chromium/chrome/browser/crypto_wallet/fragments/EditVisibleAssetsBottomSheetDialogFragment.java#L300 so we need to handle the part of adding logo for native tokens (or anything else is missing) one way or another, then it can be removed

Reply:
ah, we are missing an icon for native assets? I will double check that and perhaps will make another function to add icons in that case.